whelk-core/src/main/groovy/se/kb/libris/Normalizers.groovy

```diff
@@ -179,6 +179,33 @@ class Normalizers {
         })
     }
 
+    static DocumentNormalizer contribution() {
+        return new Normalizer({ Document doc ->
+            def (record, thing) = doc.data[GRAPH_KEY]
+            findKey(thing, 'contribution') { value, _ ->
+                List<Map> contribution = asList(value)
+                def duplicates = contribution.countBy { asList(it.agent) }.findResults { it.value > 1 ? it.key : null }
+                duplicates.each { d ->
+                    def firstIdx = contribution.findIndexOf { asList(it.agent) == d }
+                    def first = contribution[firstIdx]
+                    def roles = []
+                    contribution.removeAll {
+                        if (asList(it.agent) == d) {
+                            roles += it.role
```

Comment:
```suggestion
                            roles += asList(it.role)
```
I think?

Reply:
Exactly! See variants of the `plus()` method: https://docs.groovy-lang.org/latest/html/groovy-jdk/java/util/List.html#plus(java.lang.Object).